File: resolver/src/main/java/com/leo/sp/resolver/SpContants.java
package com.leo.sp.resolver;

import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;

public class SpContants {

    // normal constants
    public static final String CONTENT = "content://";
    public static String AUTHORITY = "";
    public static final String SEPARATOR = "/";
    public static String CONTENT_URI = "";
    public static final String TYPE_STRING = "string";
    public static final String TYPE_INT = "int";
    public static final String TYPE_LONG = "long";
    public static final String TYPE_FLOAT = "float";
    public static final String TYPE_DOUBLE = "double";
    public static final String TYPE_BOOLEAN = "boolean";
    public static final String VALUE = "value";

    public static final String NULL_STRING = "null";
    public static final String TYPE_CONTAIN = "contain";
    public static final String TYPE_CLEAN = "clean";


    public static void initAuthority(Context context) {
        AUTHORITY = getMetaValue(context, "authority");
        CONTENT_URI = CONTENT + AUTHORITY;
    }

    private static String getMetaValue(Context context, String metaKey) {
        String metaValue = null;
        if (metaKey == null) {
            return null;
        }
        try {
            ApplicationInfo appInfo = context.getPackageManager().getApplicationInfo(
                    context.getPackageName(), PackageManager.GET_META_DATA);
            metaValue = appInfo.metaData.getString(metaKey);
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        return metaValue;
    }
}

File: resolver/src/main/java/com/leo/sp/resolver/SpContants.kt
package com.leo.sp.resolver

import android.content.Context
import android.content.pm.PackageManager

object SpContants {
    const val CONTENT = "content://"
    var AUTHORITY: String? = ""
    const val SEPARATOR = "/"
    var CONTENT_URI = ""
    const val TYPE_STRING = "string"
    const val TYPE_INT = "int"
    const val TYPE_LONG = "long"
    const val TYPE_FLOAT = "float"
    const val TYPE_DOUBLE = "double"
    const val TYPE_BOOLEAN = "boolean"
    const val VALUE = "value"
    const val NULL_STRING = "null"
    const val TYPE_CONTAIN = "contain"
    const val TYPE_CLEAN = "clean"
    const val TYPE_GET_ALL = "get_all"
    const val CURSOR_COLUMN_NAME = "cursor_name"
    const val CURSOR_COLUMN_TYPE = "cursor_type"
    const val CURSOR_COLUMN_VALUE = "cursor_value"
    fun initAuthority(context: Context) {
        AUTHORITY = getMetaValue(context, "authority")
        CONTENT_URI = CONTENT + AUTHORITY
    }

    fun getMetaValue(context: Context, metaKey: String?): String? {
        var metaValue: String? = null
        if (metaKey == null) {
            return null
        }
        try {
            val appInfo = context.packageManager.getApplicationInfo(
                context.packageName, PackageManager.GET_META_DATA
            )
            metaValue = appInfo.metaData.getString(metaKey)
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }
        return metaValue
    }
}
